api/envoy/extensions/clusters/dns/v3/dns_cluster.proto

```diff
@@ -0,0 +1,88 @@
+    // is 10 times the :ref:`base_interval <envoy_v3_api_field_extensions.clusters.dns.v3.DnsCluster.RefreshRate.base_interval>`.
+    google.protobuf.Duration max_interval = 2 [(validate.rules).duration = {gt {nanos: 1000000}}];
+  }
+
+  // This value is the cluster’s DNS refresh rate. The value configured must be at least 1ms.
+  // If this setting is not specified, the
+  // value defaults to 5000ms.
+  google.protobuf.Duration dns_refresh_rate = 3 [(validate.rules).duration = {gt {nanos: 1000000}}];
+
+  // This is the cluster’s DNS refresh rate when requests are failing. If this setting is
+  // not specified, the failure refresh rate defaults to the DNS refresh rate.
+  RefreshRate dns_failure_refresh_rate = 4;
+
+  // Optional configuration for setting cluster's DNS refresh rate. If the value is set to true,
+  // cluster's DNS refresh rate will be set to resource record's TTL which comes from DNS
+  // resolution.
+  bool respect_dns_ttl = 5;
+
+  // DNS jitter causes the cluster to refresh DNS entries later by a random amount of time to avoid a
+  // stampede of DNS requests. This value sets the upper bound (exclusive) for the random amount.
+  // There will be no jitter if this value is omitted.
+  google.protobuf.Duration dns_jitter = 6 [(validate.rules).duration = {gte {}}];
+
+  // DNS resolver type configuration extension. This extension can be used to configure c-ares, apple,
+  // or any other DNS resolver types and the related parameters.
+  // For example, an object of
+  // :ref:`CaresDnsResolverConfig <envoy_v3_api_msg_extensions.network.dns_resolver.cares.v3.CaresDnsResolverConfig>`
+  // can be packed into this ``typed_dns_resolver_config``. This configuration replaces the
+  // :ref:`Cluster.dns_resolution_config <envoy_v3_api_field_config.cluster.v3.Cluster.typed_dns_resolution_config>`
+  // configuration which replaces `Cluster.dns_resolution_config <envoy_v3_api_field_config.cluster.v3.Cluster.dns_resolution_config>`.
+  // During the transition period when ``DnsCluster.typed_dns_resolver_config``, ``Cluster.dns_resolution_config``, and
+  // ``Cluster.typed_dns_resolver_config`` exists, Envoy will prefer, ``DnsCluster.typed_dns_resolver_config`` over
+  // ``Cluster.typed_dns_resolver_config`` over ``cluster.dns_resolver_config``.
+  // [#extension-category: envoy.network.dns_resolver]
+  config.core.v3.TypedExtensionConfig typed_dns_resolver_config = 7;
+
+  // The DNS IP address resolution policy. If this setting is not specified, the
+  // value defaults to
+  // :ref:`AUTO<envoy_v3_api_enum_value_extensions.clusters.dns.v3.DnsCluster.DnsLookupFamily.AUTO>`.
+  common.dns.v3.DnsLookupFamily dns_lookup_family = 8;
+
+  // Whether to do logical or strict dns resolution.
+  // This field is only considered when the ``name`` field of the ``TypedConfig`` is ``envoy.cluster.dns``.
+  DnsDiscoveryType dns_discovery_type = 9;
```

Comment:
Open to better name

Reply:
I'm fine with this, I just want to avoid something like 

https://github.com/envoyproxy/envoy/blob/64036b42e1581df185d02858d0e5d48b8aaaef97/api/envoy/extensions/wasm/v3/wasm.proto#L176-L177

where we have a bool and then deprecate it for an enum.